### PR TITLE
packaging/fedora: Merge changes from Fedora Dist-Git plus trivial fix

### PR DESCRIPTION
**Pull request by Conan-Kudo**
Fedora Dist-Git carries changes that should be merged back into the snapd reference packaging files.

In addition, a trivial change to fix a broken bundled() Provides is included as a separate commit.

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
The two tests that failed have nothing to do with these changes...

```
2018-03-05 10:44:54 Failed tasks: 2
    - google:ubuntu-14.04-64:tests/main/interfaces-content
    - linode:fedora-27-64:tests/main/media-sharing
```

File: packaging/fedora/snapd.spec [hunk 1]
```diff
@@ -441,12 +450,12 @@ autoreconf --force --install --verbose
     --enable-nvidia-biarch \
     %{?with_multilib:--with-32bit-libdir=%{_prefix}/lib} \
     --with-snap-mount-dir=%{_sharedstatedir}/snapd/snap \
-    --with-merged-usr
+    --enable-merged-usr
```
[the remark this answers is not in the document]
Comment (Conan-Kudo) [reply packaging/fedora/snapd.spec]: @bboozzoo udisks2 does `/run/media` unless it is compiled to use `/media` with `--enable-fhs-media` switch. Most don't. Debian does.

[the remark this answers is not in the document]
Comment (bboozzoo) [reply packaging/fedora/snapd.spec]: I'll push a commit to this PR that should fix the test if you don't mind.

[the remark this answers is not in the document]
Comment (bboozzoo) [reply packaging/fedora/snapd.spec]: Hmm can't push, getting `permission denied`. Probably allow edits from maintainers is disabled. @Conan-Kudo I'll open a separate a PR and you'll need to merge the master back.

**Comment by Conan-Kudo** [comment]
@bboozzoo Rebased on top of current master.

**Comment by bboozzoo** [comment]
@Conan-Kudo can you rebase again or merge master? I'm not sure why, but it's showing conflicts in `packaging/fedora/snapd.spec` now. Maybe it's because the release branch got merged back into master and it touched this file.

**Comment by Conan-Kudo** [comment]
Rebased again.

**Comment by bboozzoo** [comment]
Thanks for the update!

**Pull request merged**